tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_bitwise_xor.h

```diff
@@ -0,0 +1,36 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+#include "noc_nonblocking_api.h"
+#include "limits.h"
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+template <bool APPROXIMATION_MODE, int ITERATIONS = 8>
+inline void calculate_bitwise_xor(const uint value) {
+#pragma GCC unroll 0
+    for (int d = 0; d < ITERATIONS; d++) {
+        vInt input = dst_reg[0];
+        vInt val = input;
+        vInt v = value;
+        vInt res = val ^ v;
```

Comment:
Hi, I have a question about a note you made in the front page here:

> Bitwise XOR : Scalar values can be both positive and negative. Input Tensor needs to be positive only, because vInt doesn't support negative Integers.

VInt is the notation for signed integers, so I am unsure by the meaning here? Also I am unsure why the above needs:
`vInt val = input;`? It can be summarized as :

```
vInt input = dst_reg[0];
vInt scalar_value = value;
vInt res = input ^ scalar_value;
```

Reply:
Also Id like more elaboration on how the bitwise XOR is meant to handle the sign? From my understanding of below:
```
        v_if(res > INT_MIN && res < 0)
        {
            res = 0 - res;
            res = setsgn(res, v);
        }
```
Its the operation doing XOR, then taking the sign from the scalar value, is there a reason? Should the XOR operation then be actually done on VUints, and re-interpreted into VInt in the end instead?

Reply:
> VInt is the notation for signed integers, so I am unsure by the meaning here? 

Hi, I meant to point out that vInt gives incorrect results for a negative input range for this op.
> Bitwise XOR : Scalar values can be both positive and negative. Input Tensor needs to be positive because a negative input tensor gives incorrect results.

I've changed the description here for more clarity.

> Should the XOR operation then be actually done on VUints, and re-interpreted into VInt in the end instead?

Using vInt directly works. There is no need for vUInt conversion.